src/spatialdata/_core/spatialdata.py

```diff
@@ -537,37 +655,93 @@ def filter_by_coordinate_system(self, coordinate_system: str | list[str], filter
         ----------
         coordinate_system
             The coordinate system(s) to filter by.
-        filter_table
-            If True (default), the table will be filtered to only contain regions
+        filter_tables
+            If True (default), the tables will be filtered to only contain regions
             of an element belonging to the specified coordinate system(s).
+        include_orphan_tables
+            If True (not default), include tables that do not annotate SpatialElement(s). Only has an effect if
+            filter_tables is also set to True.
 
         Returns
         -------
         The filtered SpatialData.
         """
-        from spatialdata._core.query.relational_query import _filter_table_by_coordinate_system
+        # TODO: decide whether to add parameter to filter only specific table.
+
         from spatialdata.transformations.operations import get_transformation
 
         elements: dict[str, dict[str, SpatialElement]] = {}
         element_paths_in_coordinate_system = []
         if isinstance(coordinate_system, str):
             coordinate_system = [coordinate_system]
         for element_type, element_name, element in self._gen_elements():
-            transformations = get_transformation(element, get_all=True)
-            assert isinstance(transformations, dict)
-            for cs in coordinate_system:
-                if cs in transformations:
-                    if element_type not in elements:
-                        elements[element_type] = {}
-                    elements[element_type][element_name] = element
-                    element_paths_in_coordinate_system.append(element_name)
-
-        if filter_table:
-            table = _filter_table_by_coordinate_system(self.table, element_paths_in_coordinate_system)
+            if element_type != "tables":
+                transformations = get_transformation(element, get_all=True)
+                assert isinstance(transformations, dict)
+                for cs in coordinate_system:
+                    if cs in transformations:
+                        if element_type not in elements:
+                            elements[element_type] = {}
+                        elements[element_type][element_name] = element
+                        element_paths_in_coordinate_system.append(element_name)
+        tables = self._filter_tables(
+            filter_tables, "cs", include_orphan_tables, element_paths=element_paths_in_coordinate_system
+        )
+
+        return SpatialData(**elements, tables=tables)
+
+    def _filter_tables(
```

Comment:
I would move this function to relational query. That file will need to be refactored to make the functions more organized and set some of them public (we can do it in a follow up PR)